samples/chatbot/src/main/java/io/quarkiverse/langchain4j/sample/chatbot/ChatMemoryBean.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkiverse.langchain4j.sample.chatbot;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+
+@ApplicationScoped
+public class ChatMemoryBean implements ChatMemoryProvider {
```

Comment:
Does this memory work correctly with multiple users?

Reply:
look at the session - the Websocket Session object is your memory id.

Reply:
(Sneaky and bad)

Reply:
Ah, very sneaky indeed :)